sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/XmlSensorTest.java

```diff
@@ -76,6 +78,15 @@
 
   private final RuleKey parsingErrorCheckRuleKey = RuleKey.of(CheckRepository.REPOSITORY_KEY, parsingErrorCheckKey);
 
+  @Test
```

Comment:
One last thing. Without the fix, this test is indeed taking a lot of time, and ultimately failing (that's good). But we should also add a timeout on it, in order to fail if it takes too much time. Otherwise, if we destroy the performances again but stay linear, we won't necessarily notice it.

Add maybe a 5 or 10sec timeout on it: https://junit.org/junit4/javadoc/4.12/org/junit/Test.html

Reply:
I added 10sec timeout.